core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -897,6 +897,21 @@ protected void postImport() {
             BooleanEncodedValue roadClassLinkEnc = encodingManager.getBooleanEncodedValue(RoadClassLink.KEY);
             UrbanDensityCalculator.calcUrbanDensity(baseGraph, urbanDensityEnc, roadClassEnc, roadClassLinkEnc, residentialAreaRadius, residentialAreaSensitivity, cityAreaRadius, cityAreaSensitivity, urbanDensityCalculationThreads);
         }
+
+        if (encodingManager.hasEncodedValue(Country.KEY)
+                && encodingManager.hasEncodedValue(UrbanDensity.KEY)
+                && encodingManager.hasEncodedValue(MaxSpeed.KEY)) {
```

Comment:
This check would also be more specific with a dedicated EV. Chances are that the user will inadvertently activate the new maxspeed logic.